File: src/SIR.Client/ClientModuleBoundaries.fs
namespace SIR.Client

/// Runtime-neutral helpers shared by browser composition and client qualifications.
[<RequireQualifiedAccess>]
module ClientModuleBoundaries =
    /// Produces the canonical registry representation for a keyboard gesture.
    let canonicalGesture (gesture: string) =
        if System.String.IsNullOrWhiteSpace gesture then
            ""
        else
            gesture.Trim().ToUpperInvariant()

File: src/SIR.Client.Web/.fable-rules/SIR.Client/ClientModuleBoundaries.js

import { isNullOrWhiteSpace } from "../fable_modules/fable-library-js.5.13.0/String.js";

/**
 * Produces the canonical registry representation for a keyboard gesture.
 */
export function canonicalGesture(gesture) {
    if (isNullOrWhiteSpace(gesture)) {
        return "";
    }
    else {
        return gesture.trim().toUpperCase();
    }
}

